src/appBootstrapper.jsx

```diff
@@ -212,4 +194,21 @@ if (__DEBUG__ && module.hot) {
     module.hot.accept('./routes', () => render());
 }
 
-$(document).ready(() => render());
+$(document).ready(async () => {
+
+    // we use rawServerConfig (written by JSP) if it is present
```

Comment:
i decided to fetch configuration before we even mount the react application so that react app doesn't have to deal at all with async configuration.  it can assume config is loaded no matter what.

Reply:
i could/should probably do this by adding a top layer React component above even the router.  TODO